stdlib/public/core/UnsafeBufferPointer.swift.gyb

```diff
@@ -425,12 +460,134 @@ extension UnsafeMutableBufferPointer {
-    where S.Element == Element {
+  @discardableResult
+  public func initialize<S>(from source: S) -> (S.Iterator, Index)
+    where S:Sequence, S.Element == Element {
     return source._copyContents(initializing: self)
   }
+
+  // Warning: All buffer elements must be initialized before calling this. Note 
+  // that invoking initialize(from:) may not initialize all elements. 
+  // Assigning to part of the buffer must be done using the assign(repeating:count:) 
+  // method on the buffer’s baseAddress. 
+  @_inlineable
+  public func assign(repeating repeatedValue: Element) {
+    guard let dstStart = _position else {
+      return
+    }
+
+    dstStart.assign(repeating: repeatedValue, count: _end! - dstStart)
+  }
+
+  // Warning: All buffer elements must be initialized before calling this. Note 
+  // that invoking initialize(from:) may not initialize all elements. 
+  // Assigning to part of the buffer must be done using the assign(from:count:) 
+  // method on the buffer’s baseAddress. 
+  @_inlineable
+  @discardableResult
+  public func assign<S>(from source: S) -> (S.Iterator, Index)
+    where S:Sequence, S.Element == Element {
+    return source._copyContents(assigningTo: self)
+  }
+
+  // Warning: All buffer elements must be initialized before calling this. Note 
+  // that invoking initialize(from:) may not initialize all elements. 
+  // Assigning to part of the buffer must be done using the moveAssign(from:count:) 
+  // method on the buffer’s baseAddress. 
+  @_inlineable
+  public func moveAssign(from source: UnsafeMutableBufferPointer<Element>) {
+    _debugPrecondition(source.count >= count,
+                      "${Self}.moveAssign(from:) source does not have enough elements")
+    guard let dstStart = _position,
+          let srcStart = source._position else {
+      return
+    }
+
+    dstStart.moveAssign(from: srcStart, count: source._end! - srcStart)
+  }
+
+  @_inlineable
+  public func moveInitialize(from source: UnsafeMutableBufferPointer<Element>) {
+    _debugPrecondition(source.count >= count,
+                      "${Self}.moveInitialize(from:) source does not have enough elements")
+    guard let dstStart = _position,
+          let srcStart = source._position else {
+      return
+    }
+
+    dstStart.moveInitialize(from: srcStart, count: source._end! - srcStart)
+  }
+
+  // Warning: All buffer elements must be initialized before calling this. Note 
+  // that invoking initialize(from:) may not initialize all elements. 
+  // Deinitializing part of the buffer must be done using the deinitialize(count:) 
+  // method on the buffer’s baseAddress. 
+  @_inlineable
+  public func deinitialize() {
+    guard let dstStart = _position else {
+      return
+    }
+
+    dstStart.deinitialize(count: _end! - dstStart)
+  }
+
+% end
+
+  @_inlineable
+  public
+  func withMemoryRebound<T, Result>(
+    to type: T.Type, _ body: (${Self}<T>) throws -> Result) rethrows -> Result {
```

Comment:
Style nit: This one's tricky, but `public func` should be one line, then probably to balance out the opening `(`, you could line break and deindent before `)`--although I'm not sure about this one; check elsewhere in the codebase.

Below, where you've got `body(${Self}<T>(` and `count: ` on a separate line, line return after the opening parenthesis and indent two spaces, then put everything else on that line.